### PR TITLE
Add Ladle stories, and performed some refactoring 

### PR DESCRIPTION
**Pull request by Yukinobu-Mine**
*Issue #, if available:*

*Description of changes:*
Refactor `ChatPage`, `ChatListDrawer` and related components.

useChat
- Add `conversationError`
  - Move the error handling code to `ChatPage`
- Remove dependencies
  - `useNavigate`
  - `useTranslation`
  - `useSnackbar`

ChatMessage
- Move the action handling code to `ChatPage`
  - Submit feedback
- Move the code to get related documents to `ChatPage`
- Remove dependencies
  - `useChat`

InputChatContent
- Move the appearance control code to `ChatPage`
  - hasRegenerate / disabledRegenerate
  - hasContinue / disabledContinue
  - isLoading
- Remove dependencies
  - `useChat`

TextInputChatContent
- Move the appearance control code to `ChatPage`
  - hasRegenerate / disabledRegenerate
  - isLoading
- Remove dependencies
  - `useChat`

ChatListDrawer
- Move the action handling code to `AppContent`
  - Delete conversation
  - Update conversation title
- Move the appearance control code to `AppContent`
  - isAdmin
- Remove dependencies
  - `useNavigate`
  - `useUser`

Menu
- Move the action handling code to `AppContent`
  - Select language
  - Clear conversations
- Remove dependencies
  - `useNavigate`
  - `useConversation`

AppContent
- Accept the action handling code from `Menu` and `ChatListDrawer`
  - Select language
  - Clear conversations
  - Delete conversation
  - Update conversation title

ChatPage
- Accept the code to get related documents from `ChatMessage`
- Accept the appearance control code from `InputChatContent` and `TextInputChatContent`
- Accept the error handling code from `useChat`

Add Ladle stories
- Agent
  - InputChatContent
  - InputChatContentHasRegenerate
  - ProcessingIndicator
  - Tools
- ChatListDrawer (using msw: mock service wrapper)
  - Admin
  - NonAdmin
- ChatMessage
  - Conversation
- Helper
  - Shortcuts
- InputChatContent
  - Ideal
  - HasRegenerate
  - HasContinue
- InputText
  - Text
  - URL
  - Date
  - Number
- ListItemBot
  - MyBots
  - RecentlyUsedSharedBots
  - ApiManagement
  - Analytics
- ModalDialog
  - AddApiKey
  - ClearConversations
  - DeleteApi
  - DeleteApiKey
  - DeleteBot
  - DeleteConversation
  - Feedback
  - InstructionsSamples
  - SelectLanguage
    - Performed refactoring
  - ShareBot
- PopoverMenu
  - ChatHeader
- Progress
- Select
  - Ideal
  - Clearable
- Textarea
  - Ideal
  - NoBorder

Fixed story name
- Skelton -> Skeleton
  - Indeal -> Ideal

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: frontend/src/features/agent/components/TextInputChatContent.tsx [hunk 1]
```diff
@@ -17,25 +16,23 @@ import { BaseProps } from '../../../@types/common';
 
 type Props = BaseProps & {
   disabledSend?: boolean;
+  disabledRegenerate?: boolean;
   disabled?: boolean;
   placeholder?: string;
+  hasRegenerate: boolean;
```
Comment (wadabee) [inline frontend/src/features/agent/components/TextInputChatContent.tsx]: has より can の方がわかりやすそうかなと思ったんですが、いかがでしょうか？

File: frontend/src/components/InputChatContent.tsx [hunk 2]
```diff
@@ -37,9 +36,14 @@ import {
 
 type Props = BaseProps & {
   disabledSend?: boolean;
+  disabledRegenerate?: boolean;
+  disabledContinue?: boolean;
   disabled?: boolean;
   placeholder?: string;
   dndMode?: boolean;
+  hasRegenerate: boolean;
+  hasContinue: boolean;
```
Comment (wadabee) [inline frontend/src/components/InputChatContent.tsx]: 別のところにも書いたのですが、has より can の方がわかりやすそうかなと思ったんですが、いかがでしょうか？

File: frontend/src/pages/ChatPage.tsx [hunk 3]
```diff
@@ -395,8 +415,29 @@ const ChatPage: React.FC = () => {
...
+                                  chunkBody: chunk.content,
+                                  contentType: chunk.contentType,
+                                  sourceLink: chunk.source,
+                                  rank: chunk.rank,
+                                }));
+                              } else {
+                                // For new messages, get related documents from the api
+                                return getRelatedDocuments(message.id);
+                              }
+                            })()
```
Comment (wadabee) [inline frontend/src/pages/ChatPage.tsx]: 支障がなければuseMemoでメモ化をお願いします！

**Review by wadabee — CHANGES_REQUESTED** [review]
軽微なコメントを記載したので、ご確認をお願いしますmm

File: frontend/src/pages/ChatPage.tsx (hunk 3 above)
Comment (Yukinobu-Mine) [reply frontend/src/pages/ChatPage.tsx]: @wadabee 一応 `ChatMessage` 側で受け取った `relatedDocuments` を[メモ化している](https://github.com/aws-samples/bedrock-claude-chat/pull/485/files#diff-d9bc5cfc88dfde0d963ccfc3c4405bc742f3a029a411195706eca36aee8cb554R40)のですが、このやり方はちょっと違った感じでしょうか?
`ChatPage` 側で、`messages` 内の各メッセージの `relatedDocuments` をメモ化する必要がある場合、どのような形でメモ化するのが望ましいですか?

**Comment by Yukinobu-Mine** [comment]
@wadabee has の can へのリネームを行いました。useMemo に関して質問させて頂いたので、ご確認お願いします 🙇

**Review by wadabee — APPROVED** [review]
LGTM!!

**Pull request merged**